src/zoid/buttons/util.js

```diff
@@ -362,9 +363,18 @@ export function getButtonSize(
   }
 }
 
-export const getModal: () => Object = memoize(() => {
-  if (window.paypal?.MessagesModal) {
-    return window.paypal.MessagesModal;
+export const getModal: (
+  clientID: string,
+  merchantID: $ReadOnlyArray<string> | void
+) => Object = memoize((clientID, merchantID) => {
+  const namespace = getNamespace();
+
+  if (window[namespace]?.MessagesModal) {
+    const modal = window[namespace].MessagesModal;
+    return modal({
+      account: `client-id:${clientID}`,
+      merchantId: merchantID?.join(",") || undefined,
+    });
```

Comment:
Are we able to reorder this function to reduce some nesting and duplication?
``` ts
if (!window[namespace].MessagesModal)
    await new Promise(
        // Load the modal script
    )
}
// This now captures both scenarios and we know the modal exists at this point
const modal = window[namespace].MessagesModal;
return modal(...)
```
This helps keep a singular linear flow that doesn't have 2 spots where the modal can be created where we may forget to update one over the other in the future.